routing/edge_estimator.cpp

```diff
@@ -101,6 +102,14 @@ double CarEdgeEstimator::CalcHeuristic(m2::PointD const & from, m2::PointD const
 {
   return TimeBetweenSec(from, to, m_maxSpeedMPS);
 }
+
+double CarEdgeEstimator::GetUTurnPenalty() const
+{
+  // Adds 2 minutes penalty for U-turn. The value is quite arbitrary
+  // and needs to be properly selected after a number of real-world
+  // experiments.
+  return 2 * 60;  // seconds
```

Comment:
IMHO on big roads it's quite hard to make a U-turn due to double-solid line between lanes. On the other hand, on small roads its much easier.